contracts/auth.sol

```diff
@@ -0,0 +1,60 @@
+// This program is free software: you can redistribute it and/or modify
```

Comment:
add a resource link at the top  if you reference this contact from somewhere else